include/dlr.h

```diff
@@ -416,6 +428,69 @@ int SetDLRNumThreads(DLRModelHandle* handle, int threads);
 DLR_DLL
 int UseDLRCPUAffinity(DLRModelHandle* handle, int use);
 
+/*!
+ * \brief Creates a DLR model for TVM GraphRuntime models. Accepts graph.json and params data
+ *        from memory rather than loading from files on disk. lib.so must still be on disk
+ *        since dlopen will be used to load it.
+ * \param handle The pointer to save the model handle.
+ * \param graph String of loaded graph.json file
+ * \param lib_path Path to the TVM lib.so file on disk
+ * \param params String of binary data loaded from TVM params file
+ * \param params_len length of the params string
+ * \param dev_type Device type. Valid values are in the DLDeviceType enum in dlpack.h.
+ * \param dev_id Device ID.
+ * \return 0 for success, -1 for error. Call DLRGetLastError() to get the error message.
+ */
+DLR_DLL
+int CreateTVMModelFromMemory(DLRModelHandle* handle, const char* graph, const char* lib_path,
+                             const char* params, size_t params_len, int dev_type, int dev_id);
+
+/*!
+ * \brief Creates a DLR model for TVM GraphRuntime and VMRuntime models from specified paths
+ *        to individual files rather than a directory.
+ * \param handle The pointer to save the model handle.
+ * \param paths Paths to the model files stored in different locations
+ * \param dev_type Device type. Valid values are in the DLDeviceType enum in dlpack.h.
+ * \param dev_id Device ID.
+ * \return 0 for success, -1 for error. Call DLRGetLastError() to get the error message.
+ */
+DLR_DLL
+int CreateTVMModelFromPaths(DLRModelHandle* handle, const TVMPaths* paths, int dev_type,
```

Comment:
CreateDLRModel also accepts colon separated paths to files:
`model_path="/foo/model.so;/bar/model.json;/buzz/model.params"`.
Can we use that increase of creating a new API? 

@apivovarov

Reply:
the issue is the pathvec only contains the first path in the list. When we get the colon separated string:
https://github.com/neo-ai/neo-ai-dlr/blob/3a6d2276d00ed89f5c620a18e56947e64d4c5611/src/dlr.cc#L209-L211
We only take the first one and add it to the pathvec:
https://github.com/neo-ai/neo-ai-dlr/blob/3a6d2276d00ed89f5c620a18e56947e64d4c5611/src/dlr.cc#L204-L206

Reply:
if you pass multiple paths like: `./resnet_v1_5_50/compiled.so:./resnet_v1_5_50/compiled_model.json:./resnet_v1_5_50/compiled.params`
you get this error:
```
[00:24:29] /home/ubuntu/dlr/tests/cpp/dlr_test.cc:240: TVMError: LocalFileSystem.ListDirectory ./resnet_v1_5_50/compiled.so error: Not a directory
```
Since the ListDir call just assumes each entry in the path is a directory:
https://github.com/neo-ai/neo-ai-dlr/blob/3a6d2276d00ed89f5c620a18e56947e64d4c5611/src/dlr_common.cc#L50-L54
and then it fails when it realizes the entry is a file and not a directory
https://github.com/dmlc/dmlc-core/blob/bfad207b448480783a1f428ae3d93d87032d8349/src/io/local_filesys.cc#L101-L105

Reply:
Theres no API to check if a path is a directory in dmlc or tvm or dlr